core/services/ocr2/plugins/ocr2keeper/evm21/registry_check_pipeline.go

```diff
@@ -103,6 +103,11 @@ func (r *EvmRegistry) verifyCheckBlock(_ context.Context, checkBlock, upkeepId *
 	var ok bool
 	// verify check block number and hash are valid
 	h, ok = r.bs.queryBlocksMap(checkBlock.Int64())
+	// if this block number/hash combo exists in block subscriber, this check block and hash still exist on chain and are valid
```

Comment:
can you expand a bit here along the lines

"the block hash in block subscriber might be slightly outdated, if it doesn't match then we fetch the latest from RPC"